components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,522 @@
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.bpmn.uuf.ui;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContexts;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Class to call BPMN backend
+ */
+public class BPMNExplorerServiceImpl implements BPMNExplorerService {
+    private static final Logger log = LoggerFactory.getLogger(BPMNExplorerServiceImpl.class);
+
+    private CloseableHttpClient client;
+    private String baseUrl;
+    private String userName;
+    private String passWord;
+
+    /**
+     * Initialize the HttpClient
+     *
+     * @param host
+     * @param port
+     */
+    private void setHttpClient(String host, int port) {
+        try {
+            client = HttpClients.custom()
+                    .setSSLSocketFactory(new SSLConnectionSocketFactory(
+                            SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build()))
+                    .build();
+        } catch (NoSuchAlgorithmException | eyManagementException | KeyStoreException e) {
+            log.error("Exception : ", e);
+        }
+        baseUrl = "https://" + host + ":" + port;
+        //Need to change this part after introducing user managemnet module
+        userName = "admin";
+        passWord = "admin";
+    }
+
+    /**
+     * Return BPMN process details from deployments
+     *
+     * @param pagination
+     * @param host
+     * @param port
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public JSONArray getProcessDetails(String pagination, String host, int port, int tenantId) throws IOException {
+
+        HttpGet httpGet = null;
+        CloseableHttpResponse response = null;
+        BufferedReader rd = null;
+        String line;
+        JSONArray returnDataArray = new JSONArray();
+        String cookie;
+        int start = 0;
+
+        setHttpClient(host, port);
+
+        if (pagination.equals("")) {
+            start = 0;
```

Comment:
use "".equals(pagination) to avoid possible null pointer issue. pagination.isEmpty() is also an option if null is not a possibility